### PR TITLE
Documentation for Pluggable SCM API.

### PR DESCRIPTION
**Pull request by varshavaradarajan**

### CODE REVIEW

File: source/index.html.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ includes:
   - 150-dashboard
   - 160-pipeline-config
   - 170-version
+  - 171-pluggable-scms
```
Comment (ketan) [inline source/index.html.md]: Typo here?

File: source/includes/pluggable-scm/_02-create.md.erb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+```json
+{
+  "_links": {
+    "self": {
+      "href": "http://ci.example.com/go/api/admin/scms/foobar"
+    },
+    "doc": {
+      "href": "http://api.go.cd/#scms"
```
Comment (ketan) [inline source/includes/pluggable-scm/_02-create.md.erb]: `#pluggable-scms`?

File: source/includes/pluggable-scm/_00-index.md.erb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+```
+
+```json
+{
+  "_links": {
+    "self": {
+      "href": "http://ci.example.com/go/api/admin/scms"
+    },
+    "doc": {
+      "href": "http://api.go.cd/#scms"
```
Comment (ketan) [inline source/includes/pluggable-scm/_00-index.md.erb]: `#pluggable-scms`?

File: source/includes/pluggable-scm/_00-index.md.erb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  },
+  "_embedded": {
+    "scms": [
+      {
+        "_links": {
+          "self": {
+            "href": "http://ci.example.com/go/api/admin/scms/foo"
+          },
+          "doc": {
+            "href": "http://api.go.cd/#scms"
```
Comment (ketan) [inline source/includes/pluggable-scm/_00-index.md.erb]: `#pluggable-scms`?

File: source/includes/pluggable-scm/_00-index.md.erb [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+          }
+        },
+        "id": "912ef7c8-0d82-4a5c-8f42-0a95abe82466",
+        "name": "foo",
```
Comment (ketan) [inline source/includes/pluggable-scm/_00-index.md.erb]: Use a better fake name?

File: source/includes/pluggable-scm/_00-index.md.erb [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+          "version": "1"
+        },
+        "configuration": [
+          {
+            "key": "url",
+            "value": "git@github.com:sample/example.git"
+          },
+          {
+            "key": "username",
+            "value": "admin"
```
Comment (ketan) [inline source/includes/pluggable-scm/_00-index.md.erb]: Username/passwords don't really go well with ssh urls :) Prefer using an `http(s)` url instead?

File: source/includes/pluggable-scm/_01-show.md.erb [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      "key": "password",
+      "encrypted_value": "aSdiFgRRZ6A="
+    }
+  ]
+}
+```
+
+Lists all available pluggable scm materials, these are materials that are present in the in `cruise-config.xml`.
+
+<%= available_since('16.6.0') %>
```
Comment (ketan) [inline source/includes/pluggable-scm/_01-show.md.erb]: Update all versions to 16.7.0

File: source/includes/pluggable-scm/_03-edit.md.erb [hunk 8]
```diff
@@ -0,0 +1,101 @@
+## Edit Pluggable SCM object
```
Comment (ketan) [inline source/includes/pluggable-scm/_03-edit.md.erb]: `Update`, not `Edit`

File: source/includes/pluggable-scm/_02-create.md.erb [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+<%= available_since('16.6.0') %>
+
+<p class='http-request-heading'>HTTP Request</p>
+
+`POST /go/api/admin/scms`
+
+<p class='http-request-return-description'>Returns</p>
+
+A new SCM object.
```
Comment (ketan) [inline source/includes/pluggable-scm/_02-create.md.erb]: Please link to a section that describes the SCM object and it's nested properties, you'll find examples in other places (agents, pipeline configs etc).

Work with @maheshp so you're not duplicating efforts :)

File: source/includes/pluggable-scm/_02-create.md.erb (hunk 2 above)
Comment (varshavaradarajan) [inline source/includes/pluggable-scm/_02-create.md.erb]: You had asked me to change it from pluggable-scms to scms in an earlier conversation :)

File: source/index.html.md (hunk 1 above)
Comment (varshavaradarajan) [inline source/index.html.md]: No?

File: source/includes/pluggable-scm/_02-create.md.erb (hunk 2 above)
Comment (ketan) [inline source/includes/pluggable-scm/_02-create.md.erb]: Oh yes, I remember, now :)

File: source/index.html.md (hunk 1 above)
Comment (ketan) [inline source/index.html.md]: The file is called `_170-pluggable-scms.md.erb`, so your documentation is not being rendered in the main page. I'm fairly certain.

Also wonder why middleman does not raise a ruckus about missing partials.

**Pull request merged**